### ISSUE TITLE
After booting up, screen flickers the desktop and then goes black

### ISSUE THREAD

**Issue by Brand07** [adrgumula/oc-m3800#10]
Hello! I used your files for my M3800 using the latest version of OpenCore. Installation was successful. However, when Catalina finishes booting up the screen goes black (it’s still on just no picture). I’ve figured out that if I move the cursor around the dock the screen will flash on and I can see everything for just a moment until it goes black again. 

I tried connecting an 1920x1080 external monitor through the onboard HDMI port and Catalina immediately recognized it. However, it didn’t display anything on the external monitor but on the laptop instead as if the original issue didn’t exist.

**Comment by adrgumula** [comment]
Thanks for the info , I looked like another bug to resolve .I did not test the recent version on the 10.15, but you can try to tweak custom screen res via SwitchResX app. I've read that a few folks had to lower the screen refresh rate down to 58Hz or just lower the boot screen res to something less then 4k.

You can  try to increase /decrease  screen brightness via   f-keys  (keep pressing those keys  fast)

https://www.madrau.com/

Did you set in unlocked BIOS- System Agent (SA) Configuration - Graphics Configuration -DVMT Pre-Allocated to 160MB?

**Comment by Brand07** [comment]
> Did you set in unlocked BIOS- System Agent (SA) Configuration - Graphics Configuration -DVMT Pre-Allocated to 160MB?
Thanks for following up. I confirmed that DVMT Pre-Allocated is set to 160MB.

**Comment by Brand07** [comment]
Update: I used your same EFI folder but this time installed Big Sur instead of Catalina. The issue with the flickering screen was still present after it booted. However, I noticed that the display preferences saw the internal screen resolution as 3840 x 2160, but System information saw it had a completely different resolution. (Can't remember what it was, but I can post after work). I used SwitchResX and set a custom resolution to what was displayed in System information and it worked! The only caveat is that it's at 57 Hz instead of 60, but I'm getting somewhere.

**Comment by redwood996** [comment]
@Brand07 have you tried reset NVRAM? I did experience screen flicking in the login screen after booting up. It was gone after I reset NVRAM.

**Comment by githubme1021** [comment]
My apologies i know this is of topic but, recently I updated my mbpro 15"mid 2012 to macOS Big Sur by OpenCore Patcher. Before i was on macOS Mojave. Now I'm struggling to set up my custom resolution to 2560x1080 in SwitchresX. I managed to disable the SIP, but haven't had luck till now. Any tip?

**Comment by Brand07** [comment]
So, I took another crack at this and re-hackintoshed my Dell with this EFI you provided. Here's what I found.

I did the install with the latest version of Big Sur. Booting, installation, and setup went well. Once I got to the desktop, I experienced the screen flickering again... 

Note, the screen would be off until something on the screen would change. For example moving the mouse over the dock, or rapidly pressing the Volume Up/Volume Down keys would temporarily flicker the screen on.

I plugged in an external monitor that would allow me to see both the external monitor and the laptops internal screen without issue. This time, instead of installing SwitchResX, I downloaded EasyRes from the Mac App Store. I found this solution to be much better, personally. 

Before I changed anything with the resolution I opened system information > Display and looked to see what it said my resolution was. I used EasyRes and set the screen resolution to what System Information displayed. I then restarted the machine, cleared NVRAM from the Opencore Picker menu, and then booted into MacOS.  To my surprise, it actually worked and now I'm running the latest release of Big Sur natively at 4K 60 Hz - no more screen flickering and no more external monitors hooked up to make it work.
